diff --git a/lked b/lked
index 92c7119..638c2ad 100755
--- a/lked
+++ b/lked
@@ -388,9 +388,9 @@ if ($TARGET == linux32) then
                # next attempts a static link, whereas 10 above is a dynamic link.
                # if this fails to work, try the version 10 lines shown just above.
                set MATHLIBS="-Wl,--start-group"
-               set MATHLIBS="$MATHLIBS $mpath/libmkl_intel.a "
-               set MATHLIBS="$MATHLIBS $mpath/libmkl_sequential.a "
-               set MATHLIBS="$MATHLIBS $mpath/libmkl_core.a "
+               set MATHLIBS="$MATHLIBS $mpath/libmkl_intel.so "
+               set MATHLIBS="$MATHLIBS $mpath/libmkl_sequential.so "
+               set MATHLIBS="$MATHLIBS $mpath/libmkl_core.so "
                set MATHLIBS="$MATHLIBS -Wl,--end-group"
                breaksw
             default:
@@ -656,9 +656,9 @@ if ($TARGET == linux64) then
                set iflib=intel
                if (($GMS_FORTRAN == gfortran) || ($GMS_FORTRAN == aocc)) set iflib=gf
                set MATHLIBS="-Wl,--start-group"
-               set MATHLIBS="$MATHLIBS $mpath/libmkl_${iflib}_ilp64.a "
-               set MATHLIBS="$MATHLIBS $mpath/libmkl_sequential.a "
-               set MATHLIBS="$MATHLIBS $mpath/libmkl_core.a "
+               set MATHLIBS="$MATHLIBS $mpath/libmkl_${iflib}_ilp64.so "
+               set MATHLIBS="$MATHLIBS $mpath/libmkl_sequential.so "
+               set MATHLIBS="$MATHLIBS $mpath/libmkl_core.so "
                set MATHLIBS="$MATHLIBS -Wl,--end-group"
                if ($GMS_OPENMP == true) set MATHLIBS="$MATHLIBS -ldl"
                if (($GMS_FORTRAN == gfortran) || ($GMS_FORTRAN == aocc)) then
