diff --git a/configure b/configure
index b6acd977..5d93c5a6 100755
--- a/configure
+++ b/configure
@@ -14944,7 +14944,7 @@ then :
 else $as_nop
 
       ax_save_FLAGS=$CFLAGS
-      CFLAGS="-mtune=native"
+      CFLAGS="-mtune=$AMD_ARCH"
       cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -14970,7 +14970,7 @@ eval ax_check_compiler_flags=$ax_cv_c_flags__mtune_native
 { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $ax_check_compiler_flags" >&5
 printf "%s\n" "$ax_check_compiler_flags" >&6; }
 if test "x$ax_check_compiler_flags" = xyes; then
-	CFLAGS="$CFLAGS -mtune=native"
+	CFLAGS="$CFLAGS -mtune=$AMD_ARCH"
 else
 	:
 fi
@@ -15030,7 +15030,7 @@ then :
 else $as_nop
 
       ax_save_FLAGS=$CFLAGS
-      CFLAGS="-mtune=native"
+      CFLAGS="-mtune=$AMD_ARCH"
       cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -15056,7 +15056,7 @@ eval ax_check_compiler_flags=$ax_cv_c_flags__mtune_native
 { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $ax_check_compiler_flags" >&5
 printf "%s\n" "$ax_check_compiler_flags" >&6; }
 if test "x$ax_check_compiler_flags" = xyes; then
-	CFLAGS="$CFLAGS -mtune=native"
+	CFLAGS="$CFLAGS -mtune=$AMD_ARCH"
 else
 	:
 fi
