diff --git a/AmberTools/src/cpptraj/src/Action_GIST.cpp b/AmberTools/src/cpptraj/src/Action_GIST.cpp
index e0eaedc..a933dae 100644
--- a/AmberTools/src/cpptraj/src/Action_GIST.cpp
+++ b/AmberTools/src/cpptraj/src/Action_GIST.cpp
@@ -1,5 +1,6 @@
 #include <cmath>
 #include <cfloat> // DBL_MAX
+#include <cstdint>
 #include <algorithm>
 #include "Action_GIST.h"
 #include "CpptrajStdio.h"
diff --git a/AmberTools/src/cpptraj/src/Energy/Ecalc_Nonbond.cpp b/AmberTools/src/cpptraj/src/Energy/Ecalc_Nonbond.cpp
index 3de7a16..64c9c95 100644
--- a/AmberTools/src/cpptraj/src/Energy/Ecalc_Nonbond.cpp
+++ b/AmberTools/src/cpptraj/src/Energy/Ecalc_Nonbond.cpp
@@ -1,3 +1,4 @@
+#include <cstdint>
 #include "Ecalc_Nonbond.h"
 #include "EwaldCalc_Regular.h"
 #ifdef LIBPME
diff --git a/AmberTools/src/cpptraj/src/Energy/EwaldCalc_Decomp_LJPME.cpp b/AmberTools/src/cpptraj/src/Energy/EwaldCalc_Decomp_LJPME.cpp
index d69cba8..24fdfff 100644
--- a/AmberTools/src/cpptraj/src/Energy/EwaldCalc_Decomp_LJPME.cpp
+++ b/AmberTools/src/cpptraj/src/Energy/EwaldCalc_Decomp_LJPME.cpp
@@ -1,3 +1,4 @@
+#include <cstdint>
 #include "EwaldCalc_Decomp_LJPME.h"
 #ifdef LIBPME
 #include "../CpptrajStdio.h"
diff --git a/AmberTools/src/cpptraj/src/Energy/EwaldCalc_Decomp_PME.cpp b/AmberTools/src/cpptraj/src/Energy/EwaldCalc_Decomp_PME.cpp
index 1a3fce1..9e3b178 100644
--- a/AmberTools/src/cpptraj/src/Energy/EwaldCalc_Decomp_PME.cpp
+++ b/AmberTools/src/cpptraj/src/Energy/EwaldCalc_Decomp_PME.cpp
@@ -1,3 +1,4 @@
+#include <cstdint>
 #include "EwaldCalc_Decomp_PME.h"
 #ifdef LIBPME
 #include "../CpptrajStdio.h"
diff --git a/AmberTools/src/cpptraj/src/Energy/PME_Recip.h b/AmberTools/src/cpptraj/src/Energy/PME_Recip.h
index 328c185..5228774 100644
--- a/AmberTools/src/cpptraj/src/Energy/PME_Recip.h
+++ b/AmberTools/src/cpptraj/src/Energy/PME_Recip.h
@@ -1,6 +1,7 @@
 #ifndef INC_ENERGY_PME_RECIP_H
 #define INC_ENERGY_PME_RECIP_H
 #ifdef LIBPME
+#include <cstdint>
 #include "../helpme_standalone.h"
 #include "../Timer.h"
 #include "PME_RecipParams.h"
diff --git a/AmberTools/src/cpptraj/src/GIST_PME.cpp b/AmberTools/src/cpptraj/src/GIST_PME.cpp
index 52c145d..3b7c978 100644
--- a/AmberTools/src/cpptraj/src/GIST_PME.cpp
+++ b/AmberTools/src/cpptraj/src/GIST_PME.cpp
@@ -1,5 +1,6 @@
 #ifdef LIBPME
 #include <algorithm>
+#include <cstdint>
 #include "GIST_PME.h"
 #include "Frame.h"
 #include "CpptrajStdio.h"
diff --git a/AmberTools/src/cpptraj/src/GIST_PME.h b/AmberTools/src/cpptraj/src/GIST_PME.h
index 2f68655..402ca9e 100644
--- a/AmberTools/src/cpptraj/src/GIST_PME.h
+++ b/AmberTools/src/cpptraj/src/GIST_PME.h
@@ -2,6 +2,7 @@
 #define INC_GIST_PME_H
 #ifdef LIBPME
 #include <vector>
+#include <cstdint>
 #include "AtomMask.h"
 #include "Energy/EwaldParams.h"
 #include "Energy/PME_RecipParams.h"
