diff --git a/Makefile b/Makefile
index 6c0ad2d..ddf5728 100644
--- a/Makefile
+++ b/Makefile
@@ -44,8 +44,8 @@ endif
 
 #  Choose where you want 'make install' to put the installed binary.
 #  The normal make leaves a a link to the binary from ${BASE}/bin anyway. 
-#  PREFIX  := /usr/local
-#  INSTALL_DIR := ${PREFIX}/bin
+PREFIX  := ${out}
+INSTALL_DIR := ${PREFIX}/bin
 
 #  Orient uses some routines from Numerical Recipes (W. H. Press et al.,
 #  Numerical Recipes in Fortran) which cannot be distributed with the
@@ -120,6 +120,7 @@ test tests: force
 install: ${INSTALL_DIR}/orient
 
 ${INSTALL_DIR}/orient: ${BASE}/${DIR}/orient
+	mkdir -p ${INSTALL_DIR}
 	cp -p ${BASE}/${DIR}/orient ${INSTALL_DIR}
 
 clean:
