diff --git a/tools/dptools/setup.py b/tools/dptools/setup.py
index a7aa36e6..0106c007 100644
--- a/tools/dptools/setup.py
+++ b/tools/dptools/setup.py
@@ -1,5 +1,5 @@
 #!/usr/bin/env python3
-from distutils.core import setup
+from setuptools import setup
 
 setup(
     name="dptools",
diff --git a/tools/pythonapi/setup.py b/tools/pythonapi/setup.py
index 4188736f..424ea0ca 100644
--- a/tools/pythonapi/setup.py
+++ b/tools/pythonapi/setup.py
@@ -1,5 +1,5 @@
 #!/usr/bin/env python3
-from distutils.core import setup
+from setuptools import setup
 
 setup(
     name='pythonapi',
