diff --git a/cp2kmate.f90 b/cp2kmate.f90
index 425cc0c..af53cd8 100644
--- a/cp2kmate.f90
+++ b/cp2kmate.f90
@@ -350,7 +350,7 @@ do while(.true.)
             if (iLRIGPW==1) write(*,*) "20 Toggle using LRIGPW instead of GPW to accelerate calculation, current: Yes"
             if (iLSSCF==0) write(*,*) "21 Toggle using Linear Scaling Self Consistent Field Method, current: No"
             if (iLSSCF==1) write(*,*) "21 Toggle using Linear Scaling Self Consistent Field Method, current: Yes"
-            if (ifPBC=="XYZ") then
+            if (ifPBC==3) then
                 if (iPSOLVER==1) write(*,*) "22 Set Poisson solver, current: PERIODIC"
                 if (iPSOLVER==2) write(*,*) "22 Set Poisson solver, current: ANALYTIC"
                 if (iPSOLVER==3) write(*,*) "22 Set Poisson solver, current: MT"
