diff --git a/psi4/CMakeLists.txt b/psi4/CMakeLists.txt
index 809423041..8c04e98e0 100644
--- a/psi4/CMakeLists.txt
+++ b/psi4/CMakeLists.txt
@@ -157,15 +157,15 @@ message(STATUS "${Cyan}Using Libint ${Libint_MAX_AM_ERI}${ColourReset}: ${_loc}
 endif()
 
 # CHOOSE! this one for `make export` libint2 library (bad caps)
-#find_package(libint2 2.6 CONFIG REQUIRED) # COMPONENTS ${MAX_AM_ERI})
-#get_property(_loc TARGET Libint2::int2 PROPERTY LOCATION)
+find_package(libint2 2.6 CONFIG REQUIRED) # COMPONENTS ${MAX_AM_ERI})
+get_property(_loc TARGET Libint2::int2 PROPERTY LOCATION)
 
 # CHOOSE! this one for pure cmake libint2 library
-if(DEFINED ENV{LGTM_SRC})
-    find_package(Libint2 CONFIG REQUIRED COMPONENTS gss "eri_c4_d0_l${MAX_AM_ERI}")
-else()
-    find_package(Libint2 CONFIG REQUIRED COMPONENTS gss CXX_ho impure_sh "eri_c4_d0_l${MAX_AM_ERI}" eri_c4_d1_l2 eri_c4_d2_l2 eri_c3_d0_l4 eri_c3_d1_l3 eri_c3_d2_l3 eri_c2_d0_l4 eri_c2_d1_l3 eri_c2_d2_l3 onebody_d0_l4 onebody_d1_l3 onebody_d2_l3)
-endif()
+#if(DEFINED ENV{LGTM_SRC})
+#    find_package(Libint2 CONFIG REQUIRED COMPONENTS gss "eri_c4_d0_l${MAX_AM_ERI}")
+#else()
+#    find_package(Libint2 CONFIG REQUIRED COMPONENTS gss CXX_ho impure_sh "eri_c4_d0_l${MAX_AM_ERI}" eri_c4_d1_l2 eri_c4_d2_l2 eri_c3_d0_l4 eri_c3_d1_l3 eri_c3_d2_l3 eri_c2_d0_l4 eri_c2_d1_l3 eri_c2_d2_l3 onebody_d0_l4 onebody_d1_l3 onebody_d2_l3)
+#endif()
 get_property(_loc TARGET Libint2::int2 PROPERTY LOCATION)
 
 list(APPEND _addons ${_loc})
