diff --git a/src/external/CMakeLists.txt b/src/external/CMakeLists.txt
index 3df13b3..3f7bd62 100644
--- a/src/external/CMakeLists.txt
+++ b/src/external/CMakeLists.txt
@@ -11,19 +11,26 @@ if( GAUXC_ENABLE_HDF5 )
   if(HDF5_FOUND)
     set(GAUXC_HAS_HDF5 TRUE CACHE BOOL "" FORCE)
     message(STATUS "Enabling HDF5 Bindings")
-    message(STATUS "HighFive REPO = ${GAUXC_HIGHFIVE_REPOSITORY}")
-    message(STATUS "HighFive REV  = ${GAUXC_HIGHFIVE_REVISION}  ")
-    FetchContent_Declare( HighFive
-      GIT_REPOSITORY ${GAUXC_HIGHFIVE_REPOSITORY}
-      GIT_TAG        ${GAUXC_HIGHFIVE_REVISION}  
-    )
-    
-    set(HIGHFIVE_USE_BOOST OFF CACHE BOOL "" )
-    set(HIGHFIVE_UNIT_TESTS OFF CACHE BOOL "" )
-    set(HIGHFIVE_EXAMPLES OFF CACHE BOOL "" )
-    #set(HIGHFIVE_PARALLEL_HDF5 ON CACHE BOOL "" )
-    set(HIGHFIVE_BUILD_DOCS OFF CACHE BOOL "" )
-    FetchContent_MakeAvailable( HighFive )
+
+    find_package(HighFive)
+    if(HighFive_FOUND)
+      message(STATUS "Found installed HighFive")
+    else()
+      message(STATUS "HighFive not found, fetching from repository")
+      message(STATUS "HighFive REPO = ${GAUXC_HIGHFIVE_REPOSITORY}")
+      message(STATUS "HighFive REV  = ${GAUXC_HIGHFIVE_REVISION}  ")
+      FetchContent_Declare( HighFive
+        GIT_REPOSITORY ${GAUXC_HIGHFIVE_REPOSITORY}
+        GIT_TAG        ${GAUXC_HIGHFIVE_REVISION}  
+      )
+      
+      set(HIGHFIVE_USE_BOOST OFF CACHE BOOL "" )
+      set(HIGHFIVE_UNIT_TESTS OFF CACHE BOOL "" )
+      set(HIGHFIVE_EXAMPLES OFF CACHE BOOL "" )
+      #set(HIGHFIVE_PARALLEL_HDF5 ON CACHE BOOL "" )
+      set(HIGHFIVE_BUILD_DOCS OFF CACHE BOOL "" )
+      FetchContent_MakeAvailable( HighFive )
+    endif()
     
     target_sources( gauxc PRIVATE hdf5_write.cxx hdf5_read.cxx )
     target_link_libraries( gauxc PUBLIC HighFive )
@@ -32,4 +39,4 @@ if( GAUXC_ENABLE_HDF5 )
   endif()
 else()
   message(STATUS "Disabling HDF5 Bindings")
-endif()
+endif()
\ No newline at end of file
