diff --git a/bin/cc2molden.py b/bin/cc2molden.py
index b7d298b..8130ad3 100755
--- a/bin/cc2molden.py
+++ b/bin/cc2molden.py
@@ -5,7 +5,7 @@ from __future__ import print_function, division
 import sys
 import argparse
 
-from .. import theo_header, cclib_interface, input_options, error_handler, lib_mo
+from theodore import theo_header, cclib_interface, input_options, error_handler, lib_mo
 
 def print_warning():
     print("cc2molden.py <logfile>")
diff --git a/theodore/actions/analyze_correlations.py b/theodore/actions/analyze_correlations.py
index c3574db..e13e6d6 100755
--- a/theodore/actions/analyze_correlations.py
+++ b/theodore/actions/analyze_correlations.py
@@ -30,7 +30,7 @@ except:
 # =======================================================================
 
 try:
-    import openbabel
+    from openbabel import openbabel
     OPENBABEL=True
 except ImportError:
     print(" *** Warning: python-openbabel not found! ***")
