diff --git a/src/calculator/tblite_api.F90 b/src/calculator/tblite_api.F90
index 34e72df..4d8e416 100644
--- a/src/calculator/tblite_api.F90
+++ b/src/calculator/tblite_api.F90
@@ -376,7 +376,7 @@ contains  !> MODULE PROCEDURES START HERE
       call ceh_singlepoint(tblite%ctx,tblite%calc,mctcmol,tblite%wfn, &
       &              tblite%accuracy,verbosity)
     case (xtblvl%eeq)
-      call eeq_guess(mctcmol,tblite%calc,tblite%wfn)
+      call eeq_guess(mctcmol,tblite%calc,tblite%wfn,error)
     end select
 
     if (tblite%ctx%failed()) then
